packages/core/src/baseclient.ts

```diff
@@ -91,15 +91,8 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
 
     this._getBackend()
       .eventFromException(exception, hint)
-      .then(event => this._processEvent(event, hint, scope))
-      .then(finalEvent => {
-        // We need to check for finalEvent in case beforeSend returned null
-        eventId = finalEvent && finalEvent.event_id;
-        this._processing = false;
-      })
-      .then(null, reason => {
-        logger.error(reason);
-        this._processing = false;
+      .then(event => {
+        eventId = this.captureEvent(event, hint, scope);
```

Comment:
It looks like `eventId` is always overwritten? Even if `hint.event_id` is provided? If so, we can:

```js
return this._getBackend()
      .eventFromException(exception, hint)
      .then(event => this.captureEvent(event, hint, scope));
```

Also `this_processing = true;` is already called inside `captureEvent` so we can most likely remove it from here? Same goes for `eventId` extraction, as it's done in `captureEvent` too.

Reply:
This is tricky so I would leave it like this, because depending on if there is really a async call in between this is true or not.
`_isProcessing` should also be called before first thing since stuff in `captureX` can be async